### PR TITLE
Moving to Chisel 3

### PR DESCRIPTION
**Pull request by Thonner**
All of Patmos have been ported such that it compiles to chisel3.
Following has been commented out - and are left to the original authors to rewrite for chisel 3:
- All black boxes
- All Chisel tests

With the porting a new emulator has been build using Verilator. To build the emulator Verilator must be installed through apt-get (instructions in README).
The emulator does not yet have all features that the old emulator has. The key features will be implemented ASAP.
Missing emulator features:
- Virtual network interface with IP address
- Simulate random input from keys
- Print instruction cache statistics
- Print register values in each cycle
- Read input for UART from file <file>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,19 +101,22 @@ $(JAVATOOLSBUILDDIR)/classes/%.class: tools/java/src/%.java
...
 # Build the Chisel emulator
+#emulator: OLD
+#	-mkdir -p $(HWBUILDDIR)
+#	$(MAKE) -C hardware BOOTBUILDROOT=$(CURDIR) BOOTBUILDDIR=$(BUILDDIR) BOOTAPP=$(BOOTAPP) BOOTBIN=$(BUILDDIR)/$(BOOTAPP).bin BOARD=$(BOARD) emulator
+#	-mkdir -p $(HWINSTALLDIR)/bin
+#	cp $(HWBUILDDIR)/emulator $(HWINSTALLDIR)/bin/patemu
+
+# chisel3/verilator emulator
+CORECNTT:=$(shell lscpu | grep 'Core(s) per socket:')
 emulator:
```
Comment (schoeberl) [inline Makefile:112]: Why is the emulator target gone?

File: Makefile [hunk 2]
```diff
@@ -225,7 +228,7 @@ download: $(BUILDDIR)/$(APP).elf
 	$(INSTALLDIR)/bin/patserdow -v $(COM_PORT) $<
 
 fpgaexec: $(BUILDDIR)/$(APP).elf
-	$(INSTALLDIR)/bin/patserdow $(COM_PORT) $<
+	$(INSTALLDIR)/bin/patserdow $(COM_POpthread_mutex_testRT) $<
```
Comment (schoeberl) [inline Makefile:231]: What does COM_POpthread_mutex_testRT mean?

**Comment by schoeberl** [comment]
Otherwise LGTM and I will merge.

File: Makefile (hunk 2 above)
Comment (Thonner) [reply Makefile:231]: Whoops this is a mistake... from testing i must have copied "pthread_mutex_test" in

File: Makefile (hunk 1 above)
Comment (Thonner) [reply Makefile:112]: I kept the old target the new emulator target is below

File: Makefile (hunk 2 above)
Comment (Thonner) [reply Makefile:231]: I have reverted the mistake an pushed it to master

**Comment by schoeberl** [comment]
Ok, then I overlooked it.

Martin

**Pull request merged**